microshift.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+export LC_ALL=C.UTF-8
+export LANG=C.UTF-8
+
+source tools.sh
+source snc-library.sh
+
+BUNDLE_TYPE="microshift"
+INSTALL_DIR=crc-tmp-install-data
+SNC_PRODUCT_NAME=${SNC_PRODUCT_NAME:-crc}
+BASE_DOMAIN=${CRC_BASE_DOMAIN:-testing}
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/$ARCH/clients/ocp}
+MICROSHIFT_NVR=${MICROSHIFT_NVR:-}
+
+if [ ! -n "${MICROSHIFT_NVR}" ]; then
+    echo "Please specify MICROSHIFT_NVR, which you can get from internal brew (ex. microshift-4.12.5-202302222013.p0.g0165ac4.assembly.4.12.5.el8 )"
+    exit 1
+fi
+
+if ! grep -q -i "release 8" /etc/redhat-release
+then
+  echo "This script only works for RHEL-8"
+  exit 1
+fi
+
+echo "Check if system is registered"
+# Check the subscription status and register if necessary
+if ! sudo subscription-manager status >& /dev/null ; then
+   echo "machine must be registered using subscription-manager"
+   exit 1
+fi
+
+run_preflight_checks ${BUNDLE_TYPE}
+rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR}
+
+sudo virsh destroy ${SNC_PRODUCT_NAME} || true
+sudo virsh undefine --nvram ${SNC_PRODUCT_NAME} || true
+sudo rm -fr /var/lib/libvirt/images/${SNC_PRODUCT_NAME}.qcow2
+sudo rm -fr /var/lib/libvirt/images/microshift-installer-*.iso
+
+
+# Generate a new ssh keypair for this cluster
+# Create a 521bit ECDSA Key
+rm id_ecdsa_crc* || true
+ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+
+# This requirement is taken from https://github.com/openshift/microshift/blob/main/scripts/image-builder/configure.sh
+# Also https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/composing_a_customized_rhel_system_image/installing-composer_composing-a-customized-rhel-system-image
+# list out the dependencies and usecase.
+function configure_host {
+    sudo dnf install -y git osbuild-composer composer-cli ostree rpm-ostree \
+      cockpit-composer cockpit-machines bash-completion podman genisoimage \
+      createrepo yum-utils selinux-policy-devel jq wget lorax
+    sudo systemctl enable osbuild-composer.socket --now
+    sudo systemctl enable cockpit.socket --now
```

Comment:
Now I'm wondering if this should just be `systemctl start ...` rather than permanently enabling.